features/environment.py

```diff
@@ -22,25 +22,34 @@ class UAClientBehaveConfig:
         A string with an image name that should be used instead of building a
         fresh image for this test run.  If specified, image_clean will be set
         to False.
+    :param preserve_instance:
+        This boolean indicates that a test VM should be left running after test
+        completion. Default is to destroy the machine under test after test
+        completion.
     """
 
     prefix = "UACLIENT_BEHAVE_"
 
     # These variables are used in .from_environ() to convert the string
     # environment variable input to the appropriate Python types for use within
     # the test framework
-    boolean_options = ["image_clean"]
+    boolean_options = ["image_clean", "preserve_instance"]
     str_options = ["reuse_image"]
 
     # This variable is used in .from_environ() but also to emit the "Config
     # options" stanza in __init__
     all_options = boolean_options + str_options
 
     def __init__(
-        self, *, image_clean: bool = True, reuse_image: str = None
+        self,
+        *,
+        image_clean: bool = True,
+        reuse_image: str = None,
+        preserve_instance: bool = False
```

Comment:
I'm not entirely convinced that the pre-existing code as-written supports default-to-False perfectly; if you ever set this in your environment, then you have to either set it to "0" or you have to `unset` it entirely to disable it. Unexpectedly:

```
$ UACLIENT_BEHAVE_PRESERVE_INSTANCE=yes
# Oh, cool, it's working now
$ UACLIENT_BEHAVE_PRESERVE_INSTANCE=
$ behave
```

will set `preserve_instance` to `True`, despite the (to me) clear intent being the opposite. (You can see in `.from_environ` on this class that we "Default to True", which is why this happens.)

I don't think this is worth thinking about the semantics of (or writing code to handle) defaulting to false because a simpler alternative exists: what if we invert the config option here to `UACLIENT_BEHAVE_DESTROY_INSTANCE` which then _can_ default to True in a way that we already handle. Then:


```
$ UACLIENT_BEHAVE_DESTROY_INSTANCE=0
# Oh, cool, it's working now
$ UACLIENT_BEHAVE_DESTROY_INSTANCE=
$ behave
```

_does_ behave as we would expect (i.e. the instance is destroyed).

What do you think?